ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,292 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Third-party Imports
+from boto.ec2.elb.healthcheck import HealthCheck
+import boto.exception
+
+# Cloudify imports
+from ec2 import constants
+from ec2 import connection
+from ec2 import utils
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+
+@operation
+def creation_validation(**_):
+    """ This checks that all user supplied info is valid """
+
+    if not ctx.node.properties['resource_id']:
```

Comment:
I think this should be modified to work like this:
```
for property_key in constants.XYZ_ELB_REQUIRED_PROPS_ETC:
    utils.validate_node_property(property_key, ctx.node.properties)
instance = _get_existing_elb(utils.get_resource_id())
```
otherwise, I think it's fine